File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    answer = ''
    state = True
    for i in s:
        if i.isalpha():
            if state == True:
                answer += i.upper()
            else:
                answer += i.lower()
        elif i == " ":
            answer+= i
            state = True
            continue
        else:
            answer+=i
        state = False
    return answer

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
//11분
import java.util.Arrays;
class Solution {
    public String solution(String s) {
        boolean state = true;
        StringBuilder answer = new StringBuilder();
        for (char i : s.toCharArray()) {
            
            if (Character.isLetter(i)) {
                if (state) {
                    answer.append(Character.toUpperCase(i));
                }
                else {
                    answer.append(Character.toLowerCase(i));
                }
            }
            else if (i==' '){
                answer.append(i);
                state = true;
                continue;
            }
            else {
                answer.append(i);
            }
            state = false;
            
        }
        
        return answer.toString();
    }
}